### PR TITLE
[PSL-143] gen redundant IDs

### PR DESCRIPTION
**Pull request by mateeullahmalik**

### CODE REVIEW

File: pastel/dupe_detection.go [hunk 1]
```diff
@@ -32,6 +32,10 @@ type DDAndFingerprints struct {
 	Maxes                      *DDMaxes               `json:"maxes"`
 	Percentile                 *DDPercentiles         `json:"percentile"`
 	Score                      *DDScores              `json:"score"`
+
+	// DD-Server does not directly return these comporessed fingerprints
+	// We generate and assign to this field to avoid repeated operations
+	ZstdCompressedFingerprint []byte `json:"zstd_compressed_fingerprint"`
```
Comment (nguyen202) [inline pastel/dupe_detection.go:38]: should have `omitempty` in define it to prevent problem when un-marshall from json string

Comment (mateeullahmalik) [reply pastel/dupe_detection.go:38]: added in next pr

**Review by nguyen202 — APPROVED** [review]

**Review by a-ok123 — APPROVED** [review]

File: walletnode/services/artworkregister/task.go [hunk 2]
```diff
@@ -198,6 +213,62 @@ func (task *Task) run(ctx context.Context) error {
...
+	const digitBytes = "0123456789"
+	b := make([]byte, 4)
+	for i := range b {
+		b[i] = digitBytes[rand.Intn(len(digitBytes))]
+	}
+	randNumStr := string(b)
+
+	randNum, err := strconv.Atoi(randNumStr)
+	if err != nil {
+		return errors.Errorf("invalid random 4 bytes number: %w", err)
```
Comment (nguyen202) [inline walletnode/services/artworkregister/task.go:237]: I think randUint32() is enough for 4 bytes with higher range.
And the error due to your random generator should not pause the registration.

File: pastel/dupe_detection.go [hunk 3]
```diff
@@ -101,52 +105,65 @@ type PerceptualImageHashes struct {
...
 	if err != nil {
-		return "", errors.Errorf("GetProbeImageReply: failed to marshal dd-data: %w", err)
+		return nil, errors.Errorf("GetProbeImageReply: failed to marshal dd-data: %w", err)
 	}
 
-	resp = append(resp, separatorByte)
-	resp = append(resp, signature...)
+	res := make([]byte, base64.RawURLEncoding.EncodedLen(len(ddDataJson)))
+	encodedDdSig := make([]byte, base64.RawURLEncoding.EncodedLen(len(signature)))
+	base64.RawURLEncoding.Encode(res, ddDataJson)
```
Comment (nguyen202) [inline pastel/dupe_detection.go:157]: pasteld is using stdEncoding, so we could change to base64.StdEncoding?

**Pull request merged**